test/support/generator/get-facility-fixed-fee-generator.ts

```diff
@@ -0,0 +1,116 @@
+  constructor(protected readonly valueGenerator: RandomValueGenerator, protected readonly dateStringTransformations: DateStringTransformations) {
+    super(valueGenerator);
+  }
+
+  protected generateValues(): FacilityFixedFeeValues {
+    return {
+      fixedFeeAmount: this.valueGenerator.nonnegativeFloat({ fixed: 2 }),
+      effectiveDateInAcbs: this.valueGenerator.dateTimeString(),
+      expirationDateInAcbs: this.valueGenerator.dateTimeString(),
+      nextDueDateInAcbs: this.valueGenerator.dateTimeString(),
+      nextAccrueToDateInAcbs: this.valueGenerator.dateTimeString(),
+      segmentIdentifier: this.valueGenerator.stringOfNumericCharacters({ length: 2 }),
+      description: this.valueGenerator.string(),
+      currency: {
+        currencyCode: this.valueGenerator.string(),
+      },
+      lenderType: {
+        lenderTypeCode: this.valueGenerator.string(),
+      },
+      incomeClass: {
+        incomeClassCode: this.valueGenerator.string({ length: 3 }),
+      },
+      spreadToInvestorsIndicator: this.valueGenerator.boolean(),
+    };
+  }
+
+  protected transformRawValuesToGeneratedValues(
+    values: FacilityFixedFeeValues[],
+    { facilityIdentifier, portfolioIdentifier }: GenerateOptions,
+  ): GenerateResult {
+    const facilityFixedFeesInAcbs: AcbsGetFacilityFixedFeeResponseDto = values.map((v) => ({
+      FixedFeeAmount: v.fixedFeeAmount,
+      EffectiveDate: v.effectiveDateInAcbs,
+      ExpirationDate: v.expirationDateInAcbs,
+      NextDueDate: v.nextDueDateInAcbs,
+      NextAccrueToDate: v.nextAccrueToDateInAcbs,
+      SegmentIdentifier: v.segmentIdentifier,
+      Description: v.description,
+      Currency: {
+        CurrencyCode: v.currency.currencyCode,
+      },
+      LenderType: {
+        LenderTypeCode: v.lenderType.lenderTypeCode,
+      },
+      IncomeClass: {
+        IncomeClassCode: v.incomeClass.incomeClassCode,
+      },
+      SpreadToInvestorsIndicator: v.spreadToInvestorsIndicator,
+    }));
+
+    const facilityFixedFees: FacilityFixedFee[] = values.map((v) => ({
+      facilityIdentifier,
+      portfolioIdentifier,
+      amount: v.fixedFeeAmount,
+      effectiveDate: this.dateStringTransformations.removeTime(v.effectiveDateInAcbs),
+      expirationDate: this.dateStringTransformations.removeTime(v.expirationDateInAcbs),
+      nextDueDate: this.dateStringTransformations.removeTime(v.nextDueDateInAcbs),
+      nextAccrueToDate: this.dateStringTransformations.removeTime(v.nextAccrueToDateInAcbs),
+      period: v.segmentIdentifier,
+      description: v.description,
+      currency: v.currency.currencyCode,
+      lenderTypeCode: v.lenderType.lenderTypeCode,
+      incomeClassCode: v.incomeClass.incomeClassCode,
+      spreadToInvestorsIndicator: v.spreadToInvestorsIndicator,
+    }));
+
+    const facilityFixedFeesFromApi: GetFacilityFixedFeeResponse = facilityFixedFees;
+
+    return {
+      facilityFixedFeesInAcbs,
+      facilityFixedFees,
```

Comment:
If you remove the one-to-one mapping in the controller (see comment above) then `facilityFixedFees` won't be needed.